src/test/java/games/strategy/net/MessengerTestUtils.java

```diff
@@ -0,0 +1,25 @@
+package games.strategy.net;
+
+import javax.annotation.Nullable;
+
+/**
+ * A collection of methods useful for writing tests that use instances of {@link IMessenger}.
+ */
+public final class MessengerTestUtils {
+  private MessengerTestUtils() {}
+
+  /**
+   * Shuts down the specified messenger unconditionally. Any exception will be written to stdout.
+   *
+   * @param messenger The messenger to shut down; if {@code null}, this method does nothing.
+   */
+  public static void shutDownQuietly(final @Nullable IMessenger messenger) {
+    if (messenger != null) {
```

Comment:
I don't think such a check should be necessary, test code should be written in a way that such a field is never null

Reply:
In general, I agree with your assertion about test code not assigning `null` values to fields.  However, in this case, those fields are initialized in `setUp()`.  If `setUp()` throws an exception during execution, some fields may be left uninitialized (i.e. `null`).  Even when a `@BeforeEach` method throws, JUnit will still call the `@AfterEach` methods.  Thus, zero or more fields initialized in `setUp()` _may_ be `null` when `tearDown()` is invoked.  Thus, we have to check for `null` to avoid an NPE.

Reply:
Makes sense.
Thanks for clarifying